### ISSUE TITLE
Crash from different containers should not be removed as duplicate

### ISSUE THREAD

**Issue by marusak** [abrt/abrt#1270]
When some crash happens in container X and then the same crash happens in container Y, should only one dumdir be created or two? It can be confusing, if somebody wants to know, what crashed in a specific container.
```
# ls /var/spool/abrt 
> <nothing>
$ docker run fedora /usr/bin/bash -c "timeout -s ABRT 1 sleep 10"
> 62020d004ede102bb686266c720f7967f2e2064473e35b703e329b4a14310f6b
# ls /var/spool/abrt 
> ccpp-2017-10-19-12:40:41.11277-16927
$ docker run fedora /usr/bin/bash -c "timeout -s ABRT 1 sleep 10"
> 406e8b3801bf996f0a307831a5717f1109a64b31c76ff82919c0df929318573a
# ls /var/spool/abrt 
> ccpp-2017-10-19-12:40:41.11277-16927

# cat ccpp-2017-10-19-12:40:41.11277-16927/container_id
> 62020d004ede

**Comment by jfilak** [comment]
Good point! I have been thinking about improvements in how ABRT handles duplicates for some time now.

I don't think it is useful to keep both coredump files and other debugging files, but ABRT should not remove the dumpdir entirely - some files must be preserved and a link between the first occurrence and the other occurrence must be established. gnome-abrt / abrt-cockpit must display only the first occurrences, though, the history could be available.

The complete removal of dump directories cause problems to abrt-applet which can restart a crashed application - only it does not use the same command line to re-start the process as it uses the first captured command line - what a surprise for users!

**Comment by marusak** [comment]
Can we put some thoughts together how handling duplicates could work?

> I don't think it is useful to keep both coredump files and other debugging files, but ABRT should not remove the dumpdir entirely - some files must be preserved and a link between the first occurrence and the other occurrence must be established. 

Agreed.  Should it be done as simlinks? (simple to implement and only symlink files, that are the same) Or some more elaborated way, such as new element 'duplicate_of' and path to the original dumpdir? But in both cases there is problem if original DD gets deleted. Or any other ideas?

> gnome-abrt / abrt-cockpit must display only the first occurrences, though, the history could be available.

abrt-cockpit uses dbus API and gnome-abrt as well(?). So we only need to teach ABRT's dbus API to handle this duplicates and return problems adequately.

**Comment by xsuchy** [comment]
We can make this configurable.
